core/domain/question_services_test.py

```diff
@@ -121,6 +121,33 @@ def test_create_and_get_question_skill_link(self):
             question_services.create_new_question_skill_link(
                 self.question_id, 'skill_1')
 
+    def test_get_question_skill_links_of_skill(self):
+        question_id_2 = question_services.get_new_question_id()
+        self.save_new_question(
+            question_id_2, self.editor_id,
+            self._create_valid_question_data('ABC'))
+
+        question_id_3 = question_services.get_new_question_id()
+        self.save_new_question(
+            question_id_3, self.editor_id,
+            self._create_valid_question_data('ABC'))
+        question_services.create_new_question_skill_link(
+            self.question_id, 'skill_1')
+        question_services.create_new_question_skill_link(
+            question_id_2, 'skill_1')
+        question_services.create_new_question_skill_link(
+            question_id_3, 'skill_2')
+
+        question_skill_links = (
+            question_services.get_question_skill_links_of_skill(
+                'skill_1'))
+
+        self.assertEqual(len(question_skill_links), 2)
+        question_ids = [question_skill.question_id for question_skill
+                        in question_skill_links]
+        self.assertItemsEqual(
+            question_ids, [self.question_id, question_id_2])
+
```

Comment:
Include tests for nonexistent skills, not just happy paths. Maybe in a separate unit test.